var/spack/repos/builtin/packages/elemental/package.py

```diff
@@ -0,0 +1,112 @@
+    variant('scalapack', default=False,
+            description='Elemental: build with ScaLAPACK library')
+
+    depends_on('cmake', type='build')
+    depends_on('openblas +openmp', when='~int64_blas')
+    depends_on('metis')
+    depends_on('metis +int64', when='+int64')
+    depends_on('mpi')
+    depends_on('netlib-scalapack', when='+scalapack ~int64_blas')
+
+    def build_type(self):
+        """Returns the correct value for the ``CMAKE_BUILD_TYPE`` variable
+        :return: value for ``CMAKE_BUILD_TYPE``
+        """
+        if '+debug' in self.spec:
+            return 'Debug'
+        else:
+            return 'Release'
+
+    def cmake_args(self):
+        args = ['-DCMAKE_INSTALL_MESSAGE:STRING=LAZY',
+                '-DEL_PREFER_OPENBLAS:BOOL=TRUE',
+                '-DEL_DISABLE_SCALAPACK:BOOL={0}'.format((
+                    'ON' if '+scalapack' in self.spec else 'OFF')),
+                '-DGFORTRAN_LIB=libgfortran.so',
+                '-DBUILD_SHARED_LIBS:BOOL={0}'.format((
+                    'ON' if '+shared' in self.spec else 'OFF')),
+                '-DEL_HYBRID:BOOL={0}'.format((
+                    'ON' if '+hybrid' in self.spec else 'OFF')),
+                '-DEL_C_INTERFACE:BOOL={0}'.format((
+                    'ON' if '+c_interface' in self.spec else 'OFF')),
+                '-DINSTALL_PYTHON_PACKAGE:BOOL={0}'.format((
+                    'ON' if '+python_package' in self.spec else 'OFF')),
+                '-DEL_DISABLE_PARMETIS:BOOL={0}'.format((
+                    'ON' if '+disable_parmetis' in self.spec else 'OFF')),
+                '-DEL_DISABLE_QUAD:BOOL={0}'.format((
+                    'ON' if '+disable_quad' in self.spec else 'OFF')),
+                '-DEL_USE_64BIT_INTS:BOOL={0}'.format((
+                    'ON' if '+int64' in self.spec else 'OFF')),
+                '-DEL_USE_64BIT_BLAS_INTS:BOOL={0}'.format((
+                    'ON' if '+int64_blas' in self.spec else 'OFF'))]
+        if '+int64_blas' in self.spec:
+            args.extend(['-DEL_BLAS_SUFFIX:STRING={0}'.format((
+                '_64_' if '+int64_blas' in self.spec else '_')),
+                         '-DCUSTOM_BLAS_SUFFIX:BOOL=TRUE']),
+            if '+scalapack' in self.spec:
+                args.extend(['-DEL_LAPACK_SUFFIX:STRING={0}'.format((
+                    '_64_' if '+int64_blas' in self.spec else '_')),
+                             '-DCUSTOM_LAPACK_SUFFIX:BOOL=TRUE']),
+        else:
+            if '+scalapack' in self.spec:
+                args.extend(['-DMATH_PATHS:STRING=-L{0}/lib -L{1}/lib'.format(
+                    self.spec['openblas'].prefix, self.spec['netlib-scalapack'].prefix),
+                             '-DMATH_LIBS:STRING=-lopenblas -lscalapack'])
+            else:
+                args.extend(['-DMATH_PATHS:STRING=-L{0}/lib'.format(
+                    self.spec['openblas'].prefix),
+                             '-DMATH_LIBS:STRING=-lopenblas'])
```

Comment:
Again, this is a potential problem.  Different BLAS implemenations give different names to their blas libraries.  This package should really be set up so it can work with any BLAS.